opentelemetry-sdk/src/opentelemetry/sdk/metrics/export/exemplars.py

```diff
@@ -0,0 +1,354 @@
+        This may need to change with a different SDK implementation.
+        """
+        return set2
+
+    @property
+    def sample_set(self):
+        if self._statistical:
+            for exemplar in self._sample_set:
+                exemplar.set_sample_count(
+                    self.rand_count / len(self._sample_set)
+                )
+        return self._sample_set
+
+    def reset(self):
+        self._sample_set = []
+        self.rand_count = 0
+
+
+class MinMaxExemplarSampler(ExemplarSampler):
+    """
+        Sample the minimum and maximum measurements recorded only
+    """
+
+    def __init__(self, k: int, statistical: bool = False):
+        # K will always be 2 (min and max), and selecting min and max can never be statistical
+        super().__init__(2, statistical=False)
+        self._sample_set = []
+
+    def sample(self, exemplar: Exemplar, **kwargs):
+        self._sample_set = [
+            min(
+                self._sample_set + [exemplar],
+                key=lambda exemplar: exemplar.value,
+            ),
+            max(
+                self._sample_set + [exemplar],
+                key=lambda exemplar: exemplar.value,
+            ),
+        ]
+        if self._sample_set[0] == self._sample_set[1]:
+            self._sample_set = [self._sample_set[0]]
+
+    @property
+    def sample_set(self):
+        return self._sample_set
+
+    def merge(self, set1: List[Exemplar], set2: List[Exemplar]):
+        """
+        Assume that set2 is the latest set of exemplars.
+        For simplicity, we will just keep set2 and assume set1 has already been exported.
+        This may need to change with a different SDK implementation.
+        """
+        return set2
+
+    def reset(self):
+        self._sample_set = []
+
+
+class BucketedExemplarSampler(ExemplarSampler):
+    """
+        Randomly sample k exemplars for each bucket in the aggregator.
+
+        If `BucketedExemplarSampler` is specified to be statistical, it will add a sample_count to every exemplar it records.
+        This value will be equal to `len(bucket.exemplars) / bucket.count`, that is the number of measurements each exemplar represents.
+    """
+
+    def __init__(
+        self, k: int, statistical: bool = False, boundaries: list = None
+    ):
+        super().__init__(k)
+        self._boundaries = boundaries
+        self._sample_set = [
+            RandomExemplarSampler(k, statistical=statistical)
+            for _ in range(len(self._boundaries) + 1)
+        ]
+
+    def sample(self, exemplar: Exemplar, **kwargs):
+        bucket_index = kwargs.get("bucket_index")
+        if bucket_index is None:
+            return
+
+        self._sample_set[bucket_index].sample(exemplar)
+
+    @property
+    def sample_set(self):
+        return list(
+            itertools.chain.from_iterable(
+                [sampler.sample_set for sampler in self._sample_set]
```

Comment:
can avoid allocating a new inner list
```suggestion
                sampler.sample_set for sampler in self._sample_set
```